src/overlays/actors/ovl_En_Ge3/z_en_ge3.c

```diff
@@ -28,49 +46,386 @@ const ActorInit En_Ge3_InitVars = {
+    Actor_UpdateBgCheckInfo(play, &this->picto.actor, 40.0f, 25.0f, 40.0f, 4);
+}
+
+void EnGe3_Blink(EnGe3* this, PlayState* play) {
+    if (DECR(this->blinkTimer) == 0) {
+        this->blinkTimer = Rand_S16Offset(60, 60);
+    }
+
+    this->eyeIndex = this->blinkTimer;
+
+    if (this->eyeIndex >= 3) {
+        this->eyeIndex = 0;
+    }
+}
+
+void EnGe3_Update(Actor* thisx, PlayState* play) {
+    EnGe3* this = (EnGe3*)thisx;
 
-extern ColliderCylinderInit D_809A0DA0;
+    EnGe3_UpdateColliderAndMove(&this->picto.actor, play);
+    this->actionFunc(this, play);
+    EnGe3_Blink(this, play);
+}
 
-extern UNK_TYPE D_0600A808;
+s32 EnGe3_ValidatePictograph(PlayState* play, Actor* thisx) {
+    s32 ret = Snap_ValidatePictograph(play, thisx, PICTOGRAPH_PIRATE_GOOD, &thisx->focus.pos, &thisx->shape.rot, 10.0f,
+                                      400.0f, -1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/EnGe3_Init.s")
+    ret |= Snap_ValidatePictograph(play, thisx, PICTOGRAPH_PIRATE_TOO_FAR, &thisx->focus.pos, &thisx->shape.rot, 10.0f,
+                                   1200.0f, -1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/EnGe3_Destroy.s")
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A0070.s")
+s32 EnGe3_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnGe3* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A00F8.s")
+    switch (limbIndex) {
+        case GERUDO_RED_LIMB_VEIL:
+        case GERUDO_RED_LIMB_RIGHT_SWORD:
+        case GERUDO_RED_LIMB_LEFT_SWORD:
+            *dList = NULL;
+            return false;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A020C.s")
+        case GERUDO_RED_LIMB_HEAD:
+            rot->x += this->headRot.y;
+            // fallthrough
+        default:
+            // This is required since EnGe3 shares a skeleton with EnKaizoku; it avoids stale colours being used in the
+            // displaylists.
+            OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A024C.s")
+            switch (limbIndex) {
+                case GERUDO_RED_LIMB_NECK:
+                    break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A0350.s")
+                case GERUDO_RED_LIMB_HEAD:
+                    gDPPipeSync(POLY_OPA_DISP++);
+                    gDPSetEnvColor(POLY_OPA_DISP++, 80, 60, 10, 255);
+                    break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge3/func_809A03AC.s")
+                case GERUDO_RED_LIMB_RIGHT_SWORD:
+                case GERUDO_RED_LIMB_LEFT_SWORD:
+                    gDPPipeSync(POLY_OPA_DISP++);
+                    gDPSetEnvColor(POLY_OPA_DISP++, 140, 170, 230, 255);
+                    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
```

Comment:
Wait... but we cant get into this switch case unless we enter default and these cases were already taken above? does this code never get reached?

Do we know what this does to the sword color?

Reply:
My guess is this is just copypaste from `z_geldb`'s OverrideLimbDraw (feel free to check, I think they changed it a bit for `z_kaizoku`), to just avoid stale colours altogether, so yes, it probably is ineffective.